interface/tests/test_interface.py

```diff
@@ -656,3 +656,21 @@ def default_classmethod(cls, x):
 
 Consider changing the implementation of default_method or making these attributes part of HasDefault."""  # noqa
     assert second == expected_second
+
+
+def test_wrapped_implementation():
+    class I(Interface):  # pragma: nocover
+        def f(self, a, b, c):
+            pass
+
+    def wrapping_decorator(f):
+        @wraps(f)
+        def inner(*args, **kwargs):  # pragma: nocover
+            pass
+
+        return inner
+
+    class C(implements(I)):  # pragma: nocover
+        @wrapping_decorator
```

Comment:
Can we also test that we fail correctly here if the impl signature is incorrect?

Reply:
sure